### PR TITLE
Add postgres

### PR DESCRIPTION
**Pull request by gearnode**

### CODE REVIEW

File: postgres/templates/network-policy.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  name: {{ template "postgresql.fullname" . }}
+  labels:
+    app: {{ template "postgresql.fullname" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  podSelector:
+    matchLabels:
+      app: {{ template "postgresql.fullname" . }}
```
Comment (lukkor) [inline postgres/templates/network-policy.yaml]: You should match on release too.

File: postgres/templates/network-policy.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  ingress:
+    # Allow inbound connections
+    - ports:
+      - port: 5432
+    {{- if not .Values.networkPolicy.allowExternal }}
+      from:
+      - podSelector:
+          matchLabels:
+            {{ template "postgresql.fullname" . }}-client: "true"
```
Comment (lukkor) [inline postgres/templates/network-policy.yaml]: double quote for the key

File: postgres/templates/statefulset.yaml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            - exec pg_isready --host $POD_IP
+          initialDelaySeconds: 5
+          timeoutSeconds: 3
+          periodSeconds: 5
+      {{- if not .Values.persistence.enabled }}
+      volumes:
+      - name: empty-data
+        emptyDir: {}
+      {{- end }}
+  {{- if and .Values.persistence.enabled }}
```
Comment (lukkor) [inline postgres/templates/statefulset.yaml]: You can remove `and` in the condition.

File: postgres/templates/statefulset.yaml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            - exec pg_isready --host $POD_IP
+          initialDelaySeconds: 60
+          timeoutSeconds: 5
+          failureThreshold: 7
+        readinessProbe:
+          exec:
+            command:
+            - sh
+            - -c
+            - exec pg_isready --host $POD_IP
```
Comment (lukkor) [inline postgres/templates/statefulset.yaml]: oneline?

File: postgres/values.yaml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+  # annotations: {}
+
+## Configure resource requests and limits
+## ref: http://kubernetes.io/docs/user-guide/compute-resources/
+##
+resources:
+  requests:
+    memory: 256Mi
+    cpu: 100m
```
Comment (lukkor) [inline postgres/values.yaml]: optionnal?

File: postgres/values.yaml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    ## Enable creation of NetworkPolicy resources.
+    ##
+    enabled: false
+
+    ## The Policy model to apply. When set to false, only pods with the correct
+    ## client label will have network access to the port PostgreSQL is listening
+    ## on. When true, PostgreSQL will accept connections from any source
+    ## (with the correct destination port).
+    ##
+    allowExternal: true
```
Comment (lukkor) [inline postgres/values.yaml]: You can comment this line.

File: postgres/templates/statefulset.yaml [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+{{ toYaml .Values.resources | indent 10 }}
+        env:
+        - name: POSTGRES_USER
+          value: {{ default "postgres" .Values.postgresUser | quote }}
+        - name: PGUSER
+          value: {{ default "postgres" .Values.postgresUser | quote }}
+        - name: POSTGRES_DB
+          value: {{ default "postgres" .Values.postgresDatabase | quote }}
+        - name: POSTGRES_PASSWORD
+          {{- if and .Values.secret.enabled (not .Values.postgresPassword) }}
```
Comment (tauffredou) [inline postgres/templates/statefulset.yaml]: simplify to sercretName only

File: postgres/values.yaml (hunk 6 above)
Comment (tauffredou) [inline postgres/values.yaml]: commented by default

File: postgres/templates/network-policy.yaml [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  podSelector:
+    matchLabels:
+      app: {{ template "postgresql.fullname" . }}
+  ingress:
+    # Allow inbound connections
+    - ports:
```
[the remark this answers is not in the document]
Comment (gearnode) [reply postgres/templates/network-policy.yaml]: @tauffredou For now Kubernetes does not allow to name the port. I can add comment to increase the file clarity :)

File: postgres/values.yaml (hunk 5 above)
Comment (gearnode) [reply postgres/values.yaml]: I prefer keep an default, to ensure resource avaibility on the node. You can override this comportement by setting `resources` to `{}`.

Comment (tauffredou) [reply postgres/values.yaml]: ok but we need to be consistant

**Pull request merged**